sharding/contract_utils.py

```diff
@@ -1,13 +1,237 @@
-# [TODO]
-def call_sample_function(chain, block_number, shardId, sigIndex):
-    """Call the sample function and return signer address
+import os
+
+from ethereum.tools import tester as t
+from ethereum import abi, utils, vm
+from ethereum.messages import apply_transaction, apply_message
+from ethereum.transactions import Transaction
+import viper
+import rlp
+
+STARTGAS = 10 ** 8
+GASPRICE = 0
+
+_valmgr_ct = None
+_valmgr_code = None
+_valmgr_bytecode = None
+_valmgr_addr = None
+_valmgr_sender_addr = None
+_valmgr_tx = None
+
+
+class MessageFailed(Exception):
+
+    pass
+
+
+class TransactionFailed(Exception):
+
+    pass
+
+
+def get_valmgr_ct():
+    global _valmgr_ct, _valmgr_code
+    if not _valmgr_ct:
+        _valmgr_ct = abi.ContractTranslator(
+            viper.compiler.mk_full_signature(get_valmgr_code())
+        )
+    return _valmgr_ct
+
+
+def get_valmgr_code():
+    global _valmgr_code
+    if not _valmgr_code:
+        mydir = os.path.dirname(__file__)
+        valmgr_path = os.path.join(mydir, 'contracts/validator_manager.v.py')
+        _valmgr_code = open(valmgr_path).read()
+    return _valmgr_code
+
+
+def get_valmgr_bytecode():
+    global _valmgr_bytecode
+    if not _valmgr_bytecode:
+        _valmgr_bytecode = viper.compiler.compile(get_valmgr_code())
+    return _valmgr_bytecode
+
+
+def get_valmgr_addr():
+    global _valmgr_addr
+    if not _valmgr_addr:
+        create_valmgr_tx()
+    return _valmgr_addr
+
+
+def get_valmgr_sender_addr():
+    global _valmgr_sender_addr
+    if not _valmgr_sender_addr:
+        create_valmgr_tx()
+    return _valmgr_sender_addr
+
+
+def get_valmgr_tx():
+    global _valmgr_tx
+    if not _valmgr_tx:
+        create_valmgr_tx()
+    return _valmgr_tx
+
+
+def get_tx_rawhash(tx, network_id = None):
+    """Get a tx's rawhash.
+       Copied from ethereum.transactions.Transaction.sign
     """
-    return ''
+    UnsignedTransaction = Transaction.exclude(['v', 'r', 's'])
```

Comment:
Because it's a Class, instead of a normal variable?
However, I will move `Transaction.exclude(['v', 'r', 's'])` to the line 83 :)